plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -428,6 +428,9 @@ class producer_plugin_impl : public std::enable_shared_from_this<producer_plugin
             chain_plugin::handle_bad_alloc();
          } catch ( boost::interprocess::bad_alloc& ) {
             chain_plugin::handle_db_exhaustion();
+         } catch ( const fork_database_exception& ) {
+            wlog("Cannot recover from fork_db exception. Shutting down.");
```

Comment:
elog. Log the exception details. Add to the exception message the details you had in the log message.

Reply:
Agreed. Done.

Reply:
Add to the EOS_ASSERT the details you had in the log message.